xml_converter/src/packaging_protobin.cpp

```diff
@@ -284,3 +291,47 @@ void write_protobuf_file_per_map_id(
             &state);
     }
 }
+void _category_filter_by_depth(
+    const map<string, Category>* marker_categories,
+    int target_depth,
+    map<string, StringHierarchy>* category_hierarchy,
+    vector<string> parent_name,
+    int current_depth = 0) {
+    for (auto it = marker_categories->begin(); it != marker_categories->end(); it++) {
+        vector<string> child_name = parent_name;
+        child_name.push_back(it->first);
+        if (target_depth == current_depth) {
+            StringHierarchy category_filter;
+            category_filter.add_path(child_name, true);
+            (*category_hierarchy)[join(child_name, "_")] = category_filter;
```

Comment:
Is this an intentional strategy to overwrite the contents of the map element on a collision?
If not you can do 

```cpp
(*category_hierarchy)[<key>].add_path(child_name, true);
```

because C++ Maps will automatically create the elements if you try to read from that element.

Reply:
I had written out the steps so I was sure what was happening. This is a good consolidation.